### ISSUE TITLE
iCloud Restoration Failed for FreeOTP

### ISSUE THREAD

**Issue by jimbair** [freeotp/freeotp-ios#320]
I know this worked in the past, but just as a heads up, the iCloud backup and restore feature appears to be broken currently. In my case, I migrated from an iPhone 12 Pro -> iPhone 15 Pro, both using the iOS 17 public release (to be specific, 17.0.1 -> 17.0.2).

I know the last time I had to recover they migrated as expected, so something somewhere changed. I see several reports but many are vague, so I wanted to file another. Hopefully we can track it down and fix it.

**Comment by jimbair** [comment]
Sadly my old phone has been wiped/sold, though looking at my new phone it appears disabled. I'm honestly not sure if the setting carries over or not, so there's a good chance it wasn't enabled on my previous phone and that's why it didn't copy over (though it did previously, so I can only assume a change to the policy required opt-in and I didn't).

Either way, that's probably it if I had to guess. :) I'm not sure the best place to make that visible, but it's probably worth noting somewhere in the app so users know to enable that if they want to move 2FA tokens across devices.

Thank you for the response! I assume we can close this, but I'll leave that up to you.

**Comment by justin-stephenson** [comment]
> Sadly my old phone has been wiped/sold, though looking at my new phone it appears disabled. I'm honestly not sure if the setting carries over or not, so there's a good chance it wasn't enabled on my previous phone and that's why it didn't copy over (though it did previously, so I can only assume a change to the policy required opt-in and I didn't).
> 
> Either way, that's probably it if I had to guess. :) I'm not sure the best place to make that visible, but it's probably worth noting somewhere in the app so users know to enable that if they want to move 2FA tokens across devices.
> 
> Thank you for the response! I assume we can close this, but I'll leave that up to you.

Good point, thanks for your feedback. I agree it would be nice to state this requirement somewhere in the app. I'll think about how to do that, or if anyone sees this thread and has ideas let me know. For now I will leave this ticket open.

**Comment by justin-stephenson** [comment]
Changing the title to help others who may see this problem as well.

**Comment by jimbair** [comment]
> > Sadly my old phone has been wiped/sold, though looking at my new phone it appears disabled. I'm honestly not sure if the setting carries over or not, so there's a good chance it wasn't enabled on my previous phone and that's why it didn't copy over (though it did previously, so I can only assume a change to the policy required opt-in and I didn't).
> > Either way, that's probably it if I had to guess. :) I'm not sure the best place to make that visible, but it's probably worth noting somewhere in the app so users know to enable that if they want to move 2FA tokens across devices.
> > Thank you for the response! I assume we can close this, but I'll leave that up to you.
> 
> Good point, thanks for your feedback. I agree it would be nice to state this requirement somewhere in the app. I'll think about how to do that, or if anyone sees this thread and has ideas let me know. For now I will leave this ticket open.

Maybe have a text prompt show up if the app loads with no tokens? Something like:

"Missing your tokens after a restore? Be sure to enable Advanced Data Protection in your iCloud backup to be able to restore tokens onto your new device"

or something like that

**Comment by JATippit** [comment]
I'm confirming I did/do have Advanced Data Protection enabled and my codes did NOT restore correctly.
